src/server/api/routers/profile.ts

```diff
@@ -0,0 +1,28 @@
+import { clerkClient } from "@clerk/nextjs";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+
+const modelProcedure = publicProcedure
+
+export const profileRouter = createTRPCRouter({
+  getUserByUsername: modelProcedure.input(z.object({
+    username: z.string()
+  }))
+  .query( async({input}) => {
```

Comment:
```suggestion
  .query( async({input: { username }}) => {
     const [user] = await clerkClient.users.getUserList({
      username: [username],
    });
```

こんな書き方もできます！
これにより、複数でInputのParamを使用する際は記載するコード量が減り
可読性が良くなりそうです！